src/highlight/HighlightPromoter.ts

```diff
@@ -0,0 +1,92 @@
+import debounce from 'lodash/debounce';
+import { AppStore, getIsInitialized, SelectionArg, setSelectionAction } from '../store';
+import { MOUSE_PRIMARY } from '../constants';
+
+export type Options = {
+    getSelection: () => SelectionArg | null;
+    selectionChangeDelay?: number;
+    store: AppStore;
+};
+
+// Debounce 500ms for keyboard selection
+const SELECTION_CHANGE_DEBOUNCE = 500;
+
+export default class HighlightPromoter {
```

Comment:
Should this be `HighlightListener` since it doesn't have to be used only for promoting?